src/hyp3_isce2/insar_tops_burst.py

```diff
@@ -201,44 +197,100 @@ def insar_tops_single_burst(
+    swaths = list(set(int(granule.split('_')[2][2]) for granule in reference))
+    polarization = reference[0].split('_')[4]
+
+    log.info('Begin ISCE2 TopsApp run')
+    insar_tops_packaged(
+        reference=reference_safe,
+        secondary=secondary_safe,
+        swaths=swaths,
+        polarization=polarization,
+        azimuth_looks=azimuth_looks,
+        range_looks=range_looks,
+        apply_water_mask=apply_water_mask,
+        bucket=bucket,
+        bucket_prefix=bucket_prefix,
+    )
+    log.info('ISCE2 TopsApp run completed successfully')
+
+
+def oldest_granule_first(g1, g2):
+    if g1[14:29] <= g2[14:29]:
+        return [g1], [g2]
+    return [g2], [g1]
+
+
 def main():
     """HyP3 entrypoint for the burst TOPS workflow"""
     parser = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-
-    parser.add_argument('--bucket', help='AWS S3 bucket HyP3 for upload the final product(s)')
-    parser.add_argument('--bucket-prefix', default='', help='Add a bucket prefix to product(s)')
+    parser.add_argument('granules', type=str.split, nargs='*', help='Reference and secondary scene names')
+    parser.add_argument('--reference', type=str.split, nargs='+', help='List of reference scenes"')
+    parser.add_argument('--secondary', type=str.split, nargs='+', help='List of secondary scenes"')
     parser.add_argument(
         '--looks', choices=['20x4', '10x2', '5x1'], default='20x4', help='Number of looks to take in range and azimuth'
     )
     parser.add_argument(
-        '--apply-water-mask',
-        type=string_is_true,
-        default=False,
-        help='Apply a water body mask before unwrapping.',
-    )
-    parser.add_argument(
-        'granules',
-        type=str.split,
-        nargs='+',
-        help='Reference and secondary scene names'
+        '--apply-water-mask', type=string_is_true, default=False, help='Apply a water body mask before unwrapping.'
     )
+    parser.add_argument('--bucket', help='AWS S3 bucket HyP3 for upload the final product(s)')
+    parser.add_argument('--bucket-prefix', default='', help='Add a bucket prefix to product(s)')
 
     args = parser.parse_args()
 
-    granules = [item for sublist in args.granules for item in sublist]
-    if len(granules) != 2:
-        parser.error('No more than two granules may be provided.')
-    if len(granules[0]) != len(granules[1]) != 1:
-        parser.error('Must include 1 reference and 1 secondary.')
+    has_granules = args.granules is not None and len(args.granules) > 0
+    has_ref_sec = args.reference is not None and args.secondary is not None
+    if has_granules and has_ref_sec:
+        parser.error('Either the positional granules argument, or --reference and --secondary must be specified.')
+    elif not has_granules and not has_ref_sec:
+        parser.error('Only the positional granules argument, or --reference and --secondary must be specified.')
```

Comment:
I think these errors were in the wrong spots. Might be worth having the --reference and --secondary mentioned first too.
```suggestion
    if has_granules and has_ref_sec:
        parser.error('Provide either --reference and --secondary, or the positional granules argument, not both.')
    elif not has_granules and not has_ref_sec:
        parser.error('Either --reference and --secondary, or the positional granules argument, must be provided.')
```